server/dataset/cxg_dataset.py

```diff
@@ -343,14 +344,13 @@ def query_obs_array(self, term_name):
 
     def get_obs_names(self):
         # get the index from the meta data
-        obs = self.open_array("obs")
-        meta = json.loads(obs.meta["cxg_schema"])
+        meta = self._get_filtered_cxg_schema_metadata()
         index_name = meta["index"]
         return index_name
 
     def get_obs_index(self):
         obs = self.open_array("obs")
-        meta = json.loads(obs.meta["cxg_schema"])
+        meta = self._get_filtered_cxg_schema_metadata()
```

Comment:
not sure if it's a problem to open up the obs array twice, both in `obs = self.open_array("obs")` and again in `self._get_filtered_cxg_schema_metadata()`

i would expect tiledb can handle concurrent reads without blowing up, but it could result in performance being less efficient.